ibexa/rest
# Refactored minor issues

| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | N/A
| **Type**| improvement
| **Target version** | `v4.4`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

This PR contains a few fixes for issues found when working on different topics in REST package:
 * Use callables (two element array with object and method name) so that tools can pick up that those are callables and index the call).
 * Create exceptions where they occur, and do not create them before they actually happen.
 * Session Controller should not require CSRF token when authenticating (if you already have the credentials, CSRF won't save you)
 * Fix a typo in service declaration for FloatRange
 * Make editors reconfigure themselves automatically to not add newlines to snapshot (generated) files (via editorconfig)
 * Replaced `start/endAttribute` and `start/endValueElement` calls with `attribute` and `valueElement` respectively in `RestContent`.
 * Fix PHPUnit declaration. Make it so that defaults are not redeclared and configuration is compatible with version 9.

Can make separate PRs out of these, if needed.

**TODO**:
